### DISCUSSION TITLE
Is the `git-absorb` workflow possible with Stacked Git?

### DISCUSSION THREAD

**Discussion by Byron** [stacked-git/stgit#296] [Q&A]
[`git-absorb`](https://github.com/tummychow/git-absorb) allows to associate changes in the git index with the commit they are affecting in the current branch and pull out these hunks into individual `fixup! <hash>` commits.

Even though I didn't try it I am pretty sure that doing this on a patch queue will disassociate the branch with the patches.

Is there are way to have this kind of `git-absorb` workflow in conjunction with Stacked Git, either by clever use of `stg` or by using `stg` capabilities that I might not be aware of?

Thanks a lot for your advice or for sharing your opinion.

##### Motivation
`stg` covers me for most of my workflow. Sometimes though I find myself struggling to find the right patch to add a change to while working at the top of the stack. For my particular workflow it would already be enough to find the best-fitting patch based on changes in the index, and apply all of them there right away.

**Comment by BaruchWeka** [comment]
I prepare a commit (git add) and then do `stg push --index -p $PATCH` I didn't look at git-absorb recently to see if they can be integrated.
